src/views/mission/MissionList.tsx

```diff
@@ -0,0 +1,376 @@
+    id: 1,
+    name: '恢復健康的藥水',
+    description: '路卡斯的奶奶生病了，幫助他取得製作藥水的材料。',
+    response: '奶奶又能多活幾年了！', // complete will show this
+    level: 'A',
+    status: 'ongoing',
+    requirements: [
+      {
+        item: {
+          id: item.id,
+          amount: 1,
+          unreturnable: false,
+          metadata: rawItemMetadataToItemMetadata(item),
+          updatedAt: new Date(),
+        },
+        amount: 2,
+      },
+    ],
+    rewards: [
+      {
+        type: 'fish',
+        decimal: 18,
+        amount: '1000000000000000000',
+      },
+      {
+        type: 'item',
+        item: {
+          id: item.id,
+          amount: 1,
+          unreturnable: false,
+          metadata: rawItemMetadataToItemMetadata(item),
+          updatedAt: new Date(),
+        },
+        amount: 1,
+      },
+    ],
+  },
+  {
+    id: 1,
+    name: '恢復健康的藥水',
+    description: '路卡斯的奶奶生病了，幫助他取得製作藥水的材料。',
+    response: '奶奶又能多活幾年了！', // complete will show this
+    level: 'A',
+    status: 'ongoing',
+    requirements: [
+      {
+        item: {
+          id: item.id,
+          amount: 1,
+          unreturnable: false,
+          metadata: rawItemMetadataToItemMetadata(item),
+          updatedAt: new Date(),
+        },
+        amount: 2,
+      },
+    ],
+    rewards: [
+      {
+        type: 'fish',
+        decimal: 18,
+        amount: '1000000000000000000',
+      },
+      {
+        type: 'item',
+        item: {
+          id: item.id,
+          amount: 1,
+          unreturnable: false,
+          metadata: rawItemMetadataToItemMetadata(item),
+          updatedAt: new Date(),
+        },
+        amount: 1,
+      },
+    ],
+  },
+]
+
+const StyledMissionList = styled.div`
+  width: 100%;
+  background: ${({ theme }) => theme.colors.lightGray300};
+  color: ${({ theme }) => theme.colors.otterBlack};
+  padding: 10px 20px;
+  display: flex;
+  flex-direction: column;
+  gap: 10px;
+`
+
+type MissionFilter = 'ongoing' | 'finished'
+
+const filters: { label: string; value: MissionFilter }[] = [
+  { label: 'ongoing', value: 'ongoing' },
+  { label: 'finished', value: 'finished' },
+]
+
+const StyledHeader = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+`
+
+const StyledSwitcher = styled(Switcher)``
+
+export default function MissionList() {
+  const { t } = useTranslation('', { keyPrefix: 'mission' })
+  const [filter, setFilter] = useState<MissionFilter>('ongoing')
+  const [mission, setMission] = useState<Mission | null>(null)
+  return (
+    <StyledMissionList>
+      <StyledHeader>
+        <StyledSwitcher name="filter" value={filter} options={filters} onChange={value => setFilter(value as any)} />
+        <Note>{t('ongoingCap', { current: 4, max: 6 })}</Note>
+      </StyledHeader>
+      {missions.map(mission => (
+        <MissionCard key={mission.id} mission={mission} onClick={() => setMission(mission)} />
+      ))}
+      {mission && <NewMissionPopup mission={mission} />}
```

Comment:
we should always render the popup component, so that the popup transition will be smoother.